client/src/components/Upload/DragAndDropModal.vue

```diff
@@ -0,0 +1,71 @@
+<script setup>
+import { ref, computed } from "vue";
+import { useFileDrop } from "composables/fileDrop";
+import { useGlobalUploadModal } from "composables/globalUploadModal";
+
+const modalContentElement = ref(null);
+const { isFileOverDocument, isFileOverDropZone } = useFileDrop(modalContentElement, onDrop, true);
+
+const modalClass = computed(() => {
+    if (isFileOverDropZone.value) {
+        return "ui-drag-and-drop-modal drag-over";
+    } else {
+        return "ui-drag-and-drop-modal";
+    }
+});
+
+const { openGlobalUploadModal } = useGlobalUploadModal();
+
+function onDrop(event) {
+    console.debug(event.dataTransfer);
+
+    if (event.dataTransfer?.files?.length > 0) {
+        openGlobalUploadModal({
+            immediateUpload: true,
+            immediateFiles: event.dataTransfer.files,
+        });
+    }
+}
+</script>
+
+<template>
+    <b-modal v-model="isFileOverDocument" :modal-class="modalClass" hide-header hide-footer centered>
+        <div ref="modalContentElement" class="inner-content h-xl">Drop Files here to Upload them</div>
+    </b-modal>
+</template>
+
+<style lang="scss">
+@import "theme/blue.scss";
+
+.ui-drag-and-drop-modal {
+    .modal-content {
+        background-color: transparent;
+        border-radius: 16px;
+        border: 6px dashed;
```

Comment:
Minor but should we not use rem here instead of pixels?

Reply:
I tweaked the values so the dashed border looks good with the rounded corners, which is why I fear making it relative to the font size could potentially break this in the future.

Reply:
In the themes, we use `$border-radius-extralarge: 1rem;` or e.g. `$border-radius-large: 0.3125rem;`, so defining borders in relation to font sizes seems to be fine there.